services/authService.js

```diff
@@ -14,8 +12,30 @@ function AuthService(credentials) {
     this.credentials = credentials;
   }
 
-  this.tokenPromise = new Promise(function(resolve, reject) {
-    getAccessToken(that.credentials)
+  this.scheduleTokenRenewal(0);
+}
+
+AuthService.prototype.scheduleTokenRenewal = function(delay) {
+  var that = this;
+
+  setTimeout(function() {
+    that.tokenPromise = tokenPromiseInit(that.credentials)
+
+    that.tokenPromise
+      .then(function (tokenData) {
+        that.tokenExpiry = Math.round((tokenData.expires_in * 1000) / 2);
+        that.scheduleTokenRenewal(that.tokenExpriy);
+      })
+      .catch(function (error) {
+        console.log('SAP - Error while authenticating: ' + JSON.stringify(error));
+        that.scheduleTokenRenewal(500);
```

Comment:
this isn't ideal but i didnt have anywhere up the chain to pass an error and i didnt want to crash